sycl/test/basic_tests/image_constructors.cpp

```diff
@@ -0,0 +1,344 @@
+// RUN: %clang -std=c++11 %s -o %t1.out -lstdc++ -lOpenCL -lsycl
+// RUN: env SYCL_DEVICE_TYPE=HOST %t1.out
+// RUN: %clang -std=c++11 -fsycl %s -o %t2.out -lstdc++ -lOpenCL -lsycl
+// RUN: env SYCL_DEVICE_TYPE=HOST %t2.out
+// RUN: %CPU_RUN_PLACEHOLDER %t2.out
+// RUN: %GPU_RUN_PLACEHOLDER %t2.out
+// RUN: %ACC_RUN_PLACEHOLDER %t2.out
+//==------------------- image_constructors.cpp - SYCL buffer basic test
+//----------------==//
+// Part of the LLVM Project, under the Apache License v2.0 with LLVM Exceptions.
+// See https://llvm.org/LICENSE.txt for license information.
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+//
+//===----------------------------------------------------------------------===//
+
+#include <CL/sycl.hpp>
+#include <cassert>
+#include <iostream>
+
+void no_delete(void *) {}
+int main() {
+
+  cl::sycl::image_channel_order channelOrder =
+      cl::sycl::image_channel_order::rgbx;
+  cl::sycl::image_channel_type channelType =
+      cl::sycl::image_channel_type::signed_int32;
+  unsigned int elementSize = 12; // rgbx * i32
+  cl::sycl::range<3> r(3, 3, 3);
+  int numElems = 27;     // 3*3*3
```

Comment:
I suggest having as little hardcoded consts as possible. For example, numElems = r.size();

Reply:
added the change.